core/model/src/main/java/org/activityinfo/model/type/subform/ClassType.java

```diff
@@ -0,0 +1,104 @@
+package org.activityinfo.model.type.subform;
+/*
+ * #%L
+ * ActivityInfo Server
+ * %%
+ * Copyright (C) 2009 - 2013 UNICEF
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 3 of the 
+ * License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public 
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/gpl-3.0.html>.
+ * #L%
+ */
+
+import com.google.common.collect.Maps;
+import org.activityinfo.model.form.FormClass;
+import org.activityinfo.model.legacy.CuidAdapter;
+import org.activityinfo.model.resource.ResourceId;
+
+import java.util.Map;
+
+/**
+ * Represents type of form classes : locationType classes or activity classes.
+ * Mainly used to allow user define subform kind.
+ *
+ * @author yuriyz on 03/12/2015.
+ */
+public enum ClassType {
```

Comment:
You can have period subforms like Daily, Weekly. Or subform for class : Partner, Project, Location Type. ClassType represents those subform class types. Is the name confusing?